pkg/logql/engine.go

```diff
@@ -395,6 +426,14 @@ func (q *query) JoinSampleVector(next bool, r StepResult, stepEvaluator StepEval
 	}
 
 	if GetRangeType(q.params) == InstantType {
+		// an instant query sharded first/last_over_time can return a single vector
+		if mergeFirstLast {
```

Comment:
does `TestMappingEquivalence_Instant` need a test for the false case (existing behavior) here? Or is that adequately covered elsewhere?

Reply:
the tests here execute both the sharded and unsharded (aka not mergeFirstLast) versions of the queries, and then compare the results